src/core/mapping/base_mapper.cc

```diff
@@ -135,23 +136,34 @@ void BaseMapper::select_task_options(const Legion::Mapping::MapperContext ctx,
                                      const Legion::Task& task,
                                      TaskOptions& output)
 {
+  Task legate_task(&task, context, runtime, ctx);
 #ifdef LEGATE_USE_COLLECTIVE
-  for (uint32_t idx = 0; idx < task.regions.size(); ++idx) {
-    auto& req = task.regions[idx];
+  for (auto& store : legate_task.inputs()) {
+    if (store.is_future()) continue;
+    auto idx = store.requirement_index();
+    auto req = task.regions[idx];
+    if (req.privilege & LEGION_WRITE_PRIV) continue;
```

Comment:
input stores will always be mapped to read-only region requirements, so I don't think this check is necessary.